### PR TITLE
fix($parse): correctly escape unsafe identifier characters 

### PR DESCRIPTION
**Pull request by gkalpak**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug fix.

**What is the current behavior? (You can also link to an open issue here)**
While it is possible to use arbitrary identifier characters in Angular expressions, the RegExp that detect unsafe identifiers (in order to escape them in the generated JS code), was failing to detect them in some cases.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~

**Other information**:
This PR also includes some minor clean-up changes and fixes the `setLiteral()` tests to run in CSP mode too.

### CODE REVIEW

**Comment by lgalfaso** [comment]
LGTM

File: src/ng/parse.js [hunk 1]
```diff
@@ -1260,7 +1261,7 @@ ASTCompiler.prototype = {
   },
 
   nonComputedMember: function(left, right) {
-    var SAFE_IDENTIFIER = /[$_a-zA-Z][$_a-zA-Z0-9]*/;
+    var SAFE_IDENTIFIER = /^[$_a-zA-Z][$_a-zA-Z0-9]$/;
```
Comment (jbedard) [inline src/ng/parse.js]: Don't we need to keep the `*`? Otherwise almost everything will use the `['...']` format...

Comment (gkalpak) [inline src/ng/parse.js]: You are right. I must have accidentally removed it :innocent:

**Pull request closed without merging**